aws/resource_aws_codebuild_project.go

```diff
@@ -1000,6 +1026,21 @@ func expandProjectSourceData(data map[string]interface{}) codebuild.ProjectSourc
 		}
 	}
 
+	// Only valid for CODECOMMIT source types.
+	if sourceType == codebuild.SourceTypeCodecommit {
```

Comment:
Why is it valid only for Codecommit and not for other providers? It does work for us with GitHub when we set it manually (not through Terraform).

Reply:
No specific reason except that for some reason, I thought it was only valid for CodeCommit. Not sure where I thought I had seen that, but clearly should work with everything. AWS docs doesn't seem to have anything mentioning that it should be CodeCommit only.

Seeing as this has been merged, and released, I'm not sure what is the best way forward to resolve this.